src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContext.cs

```diff
@@ -37,6 +42,15 @@ public class TagHelperContext
         public IDictionary<string, object> AllAttributes { get; }
 
         /// <summary>
+        /// Gets the collection of items used to communicate with child <see cref="ITagHelper"/>s.
+        /// </summary>
+        /// <remarks>
+        /// This <see cref="IDictionary{string, object}"/> is copy-on-write in order to not affect parent
+        /// <see cref="ITagHelper"/>s.
+        /// </remarks>
+        public IDictionary<string, object> Items { get; }
```

Comment:
need unit tests confirming this property works as expected.  should have a `TagHelperContextTest` test class in general.